03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,25 +1,58 @@
 const Task = require('../models/task')
+const asyncWrapper = require('../middleware/async')
+const { createCustomError } = require('../errors/custom-error')
 
-const getAllTasks = (req, res) => {
-    res.send('get all tasks')
+const getAllTasks = async (req, res) => {
+    try {
+        let tasks = await Task.find()
+        res.status(200).json({ tasks })
+    } catch (err) {
+        res.status(500).json({ msg: err })
```

Comment:
```suggestion
        res.status(500).json({ msg: "something went wrong" })
```

Usually, we don't want to expose the content of our error message to clients if we don't know what the error is. For example, imaging the error said, "memory limit exceeded when trying to do X, Y, Z." Well, now we've given an attacker a clue that they're causing our service to do an infinite loop or something else unexpectedly expensive when they cause that particular bug. What if they sent 1000 of those malformed requests instead of 1? The load could take down our service! This is called a "denial of service" attack, where our service allows the client to make it work extra hard, to the extent that a malicious client can make our service work so hard that it can't provide responses to real clients anymore.

That's just one example – providing details about the state of our system to outside actors when we know it's malfunctioning is always bad, so we prefer to provide a vague and generic error anytime the server is sending a 500 response, and keep the dirty details in the logs for backend engineers to find later.